### DISCUSSION TITLE
How to locally test a new package contribution locally?

### DISCUSSION THREAD

**Discussion by UnderTheCarpet** [mason-org/mason.nvim#1449] [Q&A] [answered]
How to test a mason package (specification) locally during development?

It is not obvious to me how to contribute a new package. It is stated that a package must be present in the registry to be installable. But how do I test a package specification before opening a PR on the registries repository and getting it merged?

**Comment by williamboman** [comment, marked as answer]
Hello! Recently there was a new registry source protocol added (`file:`) that can be used to test local package definitions easily, for example:

```lua
require("mason").setup {
  registries = {
    "file:~/dev/mason-registry"
  }
}
```
